activerecord/lib/active_record/schema_migration.rb

```diff
@@ -21,7 +21,7 @@ def index_name
       end
 
       def table_exists?
-        connection.table_exists?(table_name)
+        connection.data_source_exists?(table_name)
```

Comment:
`SchemaMigration#table_exists?` is used for `SchemaMigration#create_table` and `SchemaMigration#drop_table`. It seems that should use `table_exists?` rather than `data_source_exists?`.